### ISSUE TITLE
checkStep function infinite loop

### ISSUE THREAD

**Issue by z3ronp** [casperjs/casperjs#1686]
Hi: 
i add log in Casper checkStep function as follow:

``` javascript
Casper.prototype.checkStep = function checkStep(self, onComplete) {
    "use strict";
    if (self.pendingWait || self.loadInProgress || self.navigationRequested || self.browserInitializing) {
        return;
    }
    var step = self.steps[self.step++];
    // self.log(f("Step is %s, self.step is %d", typeof step, self.step), 'debug');
    if (utils.isFunction(step)) {
        return self.runStep(step);
    }
    self.result.time = new Date().getTime() - self.startTime;
    self.log(f("Done %s steps in %dms, self.step is %d", self.steps.length, self.result.time, self.step), "info");
    self.log(f("Clear Interval %d", self.checker), "debug");
    clearInterval(self.checker);
    self.checker = null;
    self.step -= 1;
    self.emit('run.complete');
    try {
        if (utils.isFunction(onComplete)) {
            onComplete.call(self, self);
        } else if (utils.isFunction(self.options.onRunComplete)) {
            self.options.onRunComplete.call(self, self);
        }
    } catch (error) {
        self.emit('complete.error', error);
        if (!self.options.silentErrors) {
            throw error;
        }
    }
};
```

and set casper log level as debug
but application output follow logs:

> [info] [phantom] Done 52 steps in 69978ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 69998ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70018ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70038ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70058ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70078ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70098ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70118ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70138ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70158ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70178ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70199ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70219ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70239ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70259ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70279ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70299ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70319ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70339ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70359ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70379ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70399ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70419ms, self.step is 53
> [debug] [phantom] Clear Interval 92
> [info] [phantom] Done 52 steps in 70440ms, self.step is 53

if added new step, then run new step,and then output infinite logs, self.checker not changed 
It seem clearInterval not working, checkStep function in infinite loop

**Comment by kensoh** [comment]
Hi @z3ronp it looks like @entrptaher suspects this as memory issue with PhantomJS. Does the issue still happens for you, or if you try running on a machine with more RAM? How about running by driving through SlimerJS/Firefox?

**Comment by kensoh** [comment]
Hi @z3ronp not sure if this issue is resolved for you with latest CasperJS v1.1.4 and PhantomJS v2.1.1? Closing this for now but please reopen if it is still happening. Thanks for raising this!
